src/types.ts

```diff
@@ -55,12 +57,37 @@ export {
   JournalEntryLineItem,
 } from './types/journal'
 
+export type DateMode =
+  | 'DAY'
+  | 'DAY_RANGE'
+  | 'MONTH'
+  | 'MONTH_RANGE'
+  | 'QAURTER'
+  | 'YEAR'
+  | 'YEAR_TO_DATE'
+
+export type DateState = {
+  startDate: Date
+  endDate: Date
+  period: DatePeriod
+  mode: DatePickerMode // @TODO - make something more generic and unify with DatePickerMode
```

Comment:
Perhaps it's worth to unify this `DatePickerMode` with `DatePickerMode` from `DatePickerModeSelector` - I'm eager to resolve this with next PR